xray-client.js

```diff
@@ -30,27 +30,82 @@ class XrayClient {
     return { token };
   }
 
+  async importXrayJsonExecutionResults(results) {
+    return this.makeAuthorizedApiCall({
+      path: "/api/v2/import/execution",
+      method: "POST",
+      data: results,
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+  }
+
+  async importCucumberJsonExecutionResults(results) {
+    return this.makeAuthorizedApiCall({
+      path: "/api/v2/import/execution/cucumber",
+      method: "POST",
+      data: results,
+      headers: {
+        "Content-Type": "application/json",
+      },
+    });
+  }
+
+  async importJunitXmlExecutionResults(params) {
+    const {
+      xml,
+      pathParameters,
+    } = params;
+
+    const urlSearchParams = new URLSearchParams();
+    Object.entries(pathParameters).forEach(([key, value]) => {
+      if (value) {
+        urlSearchParams.append(key, value);
+      }
+    });
+
+    const junitXmlEndpoint = "/api/v2/import/execution/junit";
+    const path = `${junitXmlEndpoint}?${urlSearchParams.toString}`;
```

Comment:
`pathParameters` contain plugin `params` which are optional. If none are provided, then `urlSearchParams` will be empty in which case no params should be passed to the url.